hack/update-submodule.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Update submodules HEAD to remote branches
+
+set -euo pipefail
+
+function help() {
+    echo "Usage: update_submodules.sh [OPTIONS]... branch_name..."
+    echo "Update submodules HEAD to remote branches"
+    echo "Must be ran in repo root directory"
+    echo ""
+    echo "Options:"
+    echo "-h   Shows usage text"
+    echo "Example:"
+    echo "update_submodules.sh master release-4.6"
+}
+
+function update_branch_submodules() {
+    branch=$1
+    working_branch_name="$branch-SubmoduleUpdate$(date +%m-%d)"
+    echo Using branch $working_branch_name
+    git branch -D $working_branch_name || true
```

Comment:
It's deleting the branch if it already exists, say someone was updating the submodules twice in one day for whatever reason. Because the name is very specific there should never be a case where someone has a branch with said name that they care about for any reason other than updating the submodules

Its not a pipe, but an `or`, this is because if the branch does not exist the command will fail causing the script to exit. This needs to be done because of the addition of ` set -euo pipefail`

Reply:
>It's deleting the branch if it already exists, say someone was updating the submodules twice in one day for whatever reason. Because the name is very specific there should never be a case where someone has a branch with said name that they care about for any reason other than updating the submodules

I again don't follow the need to delete. Won't we just update the branch in that case?

>Its not a pipe, but an or, this is because if the branch does not exist the command will fail causing the script to exit. This needs to be done because of the addition of set -euo pipefail

Sorry I missed the second `|`.
What if the command fails because of other reasons? It will then be a false positive and can lead to a bad commit.
Also shouldn't we exit if the branch does not exist?